src/promptflow/promptflow/batch/_batch_inputs_processor.py

```diff
@@ -14,13 +14,14 @@
 from promptflow._utils.utils import resolve_dir_to_absolute
 from promptflow.batch._errors import EmptyInputsData, InputMappingError
 from promptflow.contracts.flow import FlowInputDefinition
+from promptflow.contracts.tool import InputDefinition
 
 
 class BatchInputsProcessor:
     def __init__(
         self,
         working_dir: Path,
-        flow_inputs: Mapping[str, FlowInputDefinition],
+        flow_inputs: Mapping[str, Union[FlowInputDefinition, InputDefinition]],
```

Comment:
Why we use both type here? No difference?

Reply:
`FlowInputDefinition` has two more properties: `is_caht_input` and 'is_chat_history'. Have converted to `FlowInputDefinition` here.